pkg/api/login_oauth.go

```diff
@@ -243,6 +249,22 @@ func (hs *HTTPServer) OAuthLogin(ctx *m.ReqContext) {
 	ctx.Redirect(setting.AppSubUrl + "/")
 }
 
+func isRoleAssignable(currentRole m.RoleType, incomingRole m.RoleType) bool {
+	// role hierarchy
+	roleHierarchy := map[m.RoleType]int{
+		m.RoleType("Viewer"): 0,
```

Comment:
FYI, noticed there are `const` for these https://github.com/mongodb-forks/grafana/blob/master/pkg/models/org_user.go#L20-L24

Reply:
> One question, how is this problem addressed in other auth providers (ie: ldap)?

From what I am able to tell is the ldap authentication depends on both the file order, ldap query order, and skips any OrgRole assignment on a user if it is already set.

Since the OIDC is returning groups in a slightly different order seems to be the main issue.

Reply:
Awesome. Updated.